### ISSUE TITLE
Accessibility: Panel aria-labelledby is broken

### ISSUE THREAD

**Issue by marenar** [OfficeDev/office-ui-fabric-react#1982]
#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes, but I'd like guidance.

Requested priority: High

Products/sites affected: All panels

#### Describe the issue:

Panels are currently unlabeled. I dug into this issue and determined that the onRenderHeader function takes in an untyped props and expects headerTextId to exist, but the props passed in are IPanelProps and headerTextId is not a property of IPanelProps. The onRenderHeader function sets the id of the header to headerTextId, which is undefined, and the dialog sets aria-labelledby to headerTextId. The end result is that there is no label for the panel.

#### Actual behavior:

Opening a panel in Narrator reads "dialog, focus on _____"

#### Expected behavior:

Opening a panel in Narrator should read "'My title' dialog, focus on _______"

**Comment by betrue-final-final** [comment]
@cschlechty, @jspurlin, circling the accessibility wagons here. This breaks all panels. Is there someone who can fix this?
